src/org/opendatakit/briefcase/ui/reused/source/ShowSourceForm.java

```diff
@@ -49,14 +50,11 @@ private ShowSourceForm(String action, boolean showReloadButton) {
 
     onReloadCallbacks.add(() -> {
       reloadButton.setEnabled(false);
-      TimerTask task = new TimerTask() {
-        public void run() {
-          reloadButton.setEnabled(true);
-        }
-      };
-      Timer timer = new Timer();
-      long delayMillis = 5000L;
-      timer.schedule(task, delayMillis);
+      this.reloadTimerElapsed = false;
+      this.reloadOperationCompleted = false;
+      Timer t = new Timer(5000, (e) -> enableReload(true, false));
```

Comment:
Nice, although two booleans are starting to seem confusing. Why not have two separate methods `markTimerElapsed()` and `markReloadCompleted()`, and one `updateReloadButton()` method?

```java
private void markReloadTimerElapsed() {
  reloadTimerElapsed = true;
  updateReloadButton();
}

private void markReloadCompleted() {
  reloadOperationCompleted = true;
  updateReloadButton();
}

private void updateReloadButton() {
  if (reloadTimerElapsed && reloadOperationCompleted)
    reloadButton.setEnabled(true);
}
```

Reply:
I agree that the two booleans and their surrounding logic were beginning to look complex. Your suggested approach offers greater clarity through the use of descriptive method names - I like it!